flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -206,7 +209,7 @@ public void checkClusterEmpty() throws IOException, YarnException {
 			}
 		}
 
-		flinkConfiguration = new org.apache.flink.configuration.Configuration();
+		flinkConfiguration = GlobalConfiguration.loadConfiguration();
```

Comment:
Why do we have to load the configuration from disk and cannot use a vanilla `Configuration` object?

Reply:
In secure mode, the test will append keytab and principal option to the on-disk flink-conf.yaml before starting. So using a vanilla Configuration object is actually one of the issues why the current test code is broken.

Reply:
Can't we read the vanilla configuration in `YarnTestBase#start:474` then configure the keytab and the principal and write this `Configuration` out via `BootstrapTool#writeConfiguration`? We could then simply set `flinkConfiguration` to this `Configuration`. What do you think?

Reply:
Done, please take another look.